### ISSUE TITLE
Validate warping model

### ISSUE THREAD

**Issue by zero-nnkn** [geyuying/PF-AFN#79]
When I try to train PB and PF models, I see the need to train 2 stages for each model, stage 2 (e2e) will take the checkpoint of the warp model trained from stage 1. But, I don't see you using validation when training the warping model (stage 1). Can I use part of the training set for validation (use val loss) to choose checkpoints for the next stage?
